pkg/gadgets/trace/bind/tracer/tracer_test.go

```diff
@@ -0,0 +1,659 @@
+			},
+		},
+		"pid_filter_match": {
+			getTracerConfig: func(info *utilstest.RunnerInfo) *tracer.Config {
+				return &tracer.Config{
+					TargetPid: int32(info.Pid),
+					// It's difficult to only test the PID filter since other
+					// test events are generated from the same PID on this
+					// test suite
+					MountnsMap: utilstest.CreateMntNsFilterMap(t, info.MountNsID),
+				}
+			},
+			generateEvent: bindSocketFn("127.0.0.1", unix.AF_INET, unix.SOCK_STREAM, 0),
+			validateEvent: func(t *testing.T, info *utilstest.RunnerInfo, port uint16, events []types.Event) {
+				if len(events) != 1 {
+					t.Fatalf("Wrong number of events received %d, expected 1", len(events))
+				}
+			},
+		},
+		"pid_filter_no_match": {
+			getTracerConfig: func(info *utilstest.RunnerInfo) *tracer.Config {
+				return &tracer.Config{
+					TargetPid:  int32(1),
+					MountnsMap: utilstest.CreateMntNsFilterMap(t, info.MountNsID),
+				}
+			},
+			generateEvent: bindSocketFn("127.0.0.1", unix.AF_INET, unix.SOCK_STREAM, 0),
+			validateEvent: utilstest.ExpectNoEvent[types.Event, uint16],
+		},
+		"target_ports_filter_match": {
+			getTracerConfig: func(info *utilstest.RunnerInfo) *tracer.Config {
+				return &tracer.Config{
+					MountnsMap:  utilstest.CreateMntNsFilterMap(t, info.MountNsID),
+					TargetPorts: []uint16{5555},
+				}
+			},
+			generateEvent: bindSocketFn("127.0.0.1", unix.AF_INET, unix.SOCK_STREAM, 5555),
+			validateEvent: func(t *testing.T, info *utilstest.RunnerInfo, port uint16, events []types.Event) {
+				if len(events) != 1 {
+					t.Fatalf("Wrong number of events received %d, expected 1", len(events))
+				}
+			},
+		},
+		"target_ports_filter_no_match": {
+			getTracerConfig: func(info *utilstest.RunnerInfo) *tracer.Config {
+				return &tracer.Config{
+					MountnsMap:  utilstest.CreateMntNsFilterMap(t, info.MountNsID),
+					TargetPorts: []uint16{5555},
+				}
+			},
+			generateEvent: bindSocketFn("127.0.0.1", unix.AF_INET, unix.SOCK_STREAM, 5556),
+			validateEvent: utilstest.ExpectNoEvent[types.Event, uint16],
+		},
+		"target_ports_filter_multiple_ports": {
+			getTracerConfig: func(info *utilstest.RunnerInfo) *tracer.Config {
+				return &tracer.Config{
+					MountnsMap:  utilstest.CreateMntNsFilterMap(t, info.MountNsID),
+					TargetPorts: []uint16{5555, 5556, 5557, 5558},
+				}
+			},
+			generateEvent: func() (uint16, error) {
+				// Generate 5 events but only 4 should be captured
+				ports := []uint16{5555, 5556, 5557, 5558, 5559}
+				var err error
+				for _, port := range ports {
+					_, err = bindSocket("127.0.0.1", unix.AF_INET, unix.SOCK_STREAM, int(port))
+					if err != nil {
+						return 0, err
+					}
+				}
+
+				return 0, nil
+			},
+			validateEvent: func(t *testing.T, info *utilstest.RunnerInfo, port uint16, events []types.Event) {
+				if len(events) != 4 {
+					t.Fatalf("Wrong number of events received %d, expected 1", len(events))
+				}
+			},
+		},
+		"ignore_errors_false": {
```

Comment:
Do you think it is worth making a test that combines filters? For instance:
```golang
      return &tracer.Config{
	      MountnsMap:  utilstest.CreateMntNsFilterMap(t, info.MountNsID),
	      TargetPorts: []uint16{5555},
	      IgnoreErrors: true,
      }
```

Reply:
I think it's too much. Testing each of them independently should be good enough for this.